### ISSUE TITLE
Unable to Run M1 Apple Laptop

### ISSUE THREAD

**Issue by gsilvers** [leon-thomm/Ryven#51]
Hello,

I have a new Mac M1 laptop. I've cloned the repo and pip installed the requirements but when executing Ryven.py I get the following:

==========================================================================================
;python3 Ryven.py                                                      (base)
objc[21131]: Class RunLoopModeTracker is implemented in both /opt/anaconda3/lib/python3.8/site-packages/PySide2/Qt/lib/QtCore.framework/Versions/5/QtCore (0x15951b288) and /opt/anaconda3/lib/libQt5Core.5.9.7.dylib (0x163b48a80). One of the two will be used. Which one is undefined.
QObject::moveToThread: Current thread (0x7fea9767a1c0) is not the object's thread (0x7fea9744de10).
Cannot move to target thread (0x7fea9767a1c0)

You might be loading two sets of Qt binaries into the same process. Check that all plugins are compiled against the right Qt binaries. Export DYLD_PRINT_LIBRARIES=1 and check that only one set of binaries are being loaded.
qt.qpa.plugin: Could not load the Qt platform plugin "cocoa" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: cocoa, minimal, offscreen.

fish: 'python3 Ryven.py' terminated by signal SIGABRT (Abort)

==========================================================================================

I've read issues for other projects that the version of cv was bad, I tried opencv-python==4.2.0.32 to no avail. I do have QT installed assuming this may be some sort of issue specific to the new M1 Macs?

**Reproduction**
CD /Ryven
python3 Ryven.py

**System, version**
 - OS: macOS
 - Version 11.2.1

**Comment by dmsmiley** [comment]
I am having the same issues on my M1 as well. I've tried the usual opencv uninstall, but with no luck. Did you find a solution?

**Comment by leon-thomm** [comment]
I don't have access to an M1 machine rn unfortunately. Ryven does not move any objects between threads, and the application doesn't use OpenCV, so this seems to be a PySide internal thing. While I couldn't find a bugreport describing the same errors, there have been issues reported with Big Sur recently. Maybe [this](https://bugreports.qt.io/browse/PYSIDE-1444?focusedCommentId=551940&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-551940) might help.

**Comment by dmsmiley** [comment]
Yeah, this is no bueno. I downgraded to PySide 5.13, which fixes this issue, but now I'm getting the TypeError:'Shiboken.ObjectType' object is not iterable, which can only be solved by upgrading to PySide 5.14, which gives me the original problem. C'est la vie.

**Comment by malteschade** [comment]
I created a new conda x86 env with Python 3.8.8 and only installed the requirements on my M1 MacBook Air. This at least fixed the shown error, although afterwards Ryven stopped loading and got stuck on an empty white window. Also no error message in the console.

**Comment by V1ki** [comment]
Same issues ,  fix it by using : 
```
 export QT_MAC_WANTS_LAYER=1
```
https://github.com/leon-thomm/Ryven/issues/40#issuecomment-799384371

**Comment by winglight** [comment]
> export QT_MAC_WANTS_LAYER=1

I tried this command but the same error happend.

Is it neccessary to install qt5 seperatly?

**Comment by WillAdams** [comment]
I'm having similar difficulties on an Intel Mac:

(base) williamadams@MacBook-Pro ~ % ryven
objc[82758]: Class RunLoopModeTracker is implemented in both /Users/williamadams/opt/anaconda3/lib/python3.8/site-packages/PySide2/Qt/lib/QtCore.framework/Versions/5/QtCore (0x11e54d288) and /Users/williamadams/opt/anaconda3/lib/libQt5Core.5.9.7.dylib (0x1279dca80). One of the two will be used. Which one is undefined.
QObject::moveToThread: Current thread (0x7f96912a6c60) is not the object's thread (0x7f9691122650).
Cannot move to target thread (0x7f96912a6c60)

You might be loading two sets of Qt binaries into the same process. Check that all plugins are compiled against the right Qt binaries. Export DYLD_PRINT_LIBRARIES=1 and check that only one set of binaries are being loaded.
qt.qpa.plugin: Could not load the Qt platform plugin "cocoa" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: cocoa, minimal, offscreen.

**Comment by tristanryerparke** [comment]
@WillAdams I'm having the same issue

**Comment by lidanqing123** [comment]
您好！您的邮件已收到！我查阅后，尽快回复您！
